### PR TITLE
Add district admins

### PR DESCRIPTION
**Pull request by gglin**
Partially based on https://github.com/Clever/clever-ruby/pull/63 and https://github.com/Clever/clever-ruby/pull/69, though taking a slightly different approach (not sure if the district_admins API has changed since https://github.com/Clever/clever-ruby/pull/63)

### CODE REVIEW

File: test/integration/api_operations/list_test.rb [hunk 1]
```diff
@@ -4,12 +4,17 @@ def resource_name(resource)
   resource.shortname
 end
 
-def resources
+# The DistrictAdmin resource does not support the count or limit parameters
+def resources_wo_district_admin
```
Comment (ruddzw) [inline test/integration/api_operations/list_test.rb]: Maybe a more descriptive name? `limitable_countable_resources` or something?

File: lib/clever-ruby/api_operations/page.rb [hunk 2]
```diff
@@ -66,6 +67,29 @@ def last(num = nil)
...
+      # @returns response with each elem in [:data] containing a hash with :data and :uri keys,
+      #   unless already in that format
+      # @example
+      #   input:
+      #     {data: [{id: 1, email: 'admin1@school.edu'},
+      #             {id: 2, email: 'admin2@school.edu'}]}
+      #   output:
+      #     {data: [{data: {id: 1, email: 'admin1@school.edu'}, uri: '/v1.1/district_admins/1'},
+      #             {data: {id: 2, email: 'admin2@school.edu'}, uri: '/v1.1/district_admins/2'}]}
+      def standardize_response(response)
```
Comment (ruddzw) [inline lib/clever-ruby/api_operations/page.rb]: According to the API explorer, there is an argument you can pass to the `/district_admins` URL called `show_links` that makes it return it in this format. Although it looks like you put some real work into this, I think that would be much simpler.

File: lib/clever-ruby/district_admin.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    @uri = 'district_admins'
+    @plural = 'district_admins'
+    @linked_resources = []
+
+    # Optional attributes
+    # @see Clever::CleverObject.optional_attributes
+    # @api private
+    # @return [Array]
+    def optional_attributes
+      [:name, :title]
```
Comment (ruddzw) [inline lib/clever-ruby/district_admin.rb:33]: I was looking at other files in the same directory to see what they had for this and it appears that it thinks School Admins need a name but don't need an email, but that doesn't jive with what the clever data schema table said. I wonder what's up with that?

**Review by ruddzw — CHANGES_REQUESTED** [review]
Overall looks like this is pretty simple, most of the changes are just for tests. I do notice that the other PRs had so many more changes than this -- do you think that's mostly tests, and would we have to beef up the tests here if we wanted to get this merged in?

File: lib/clever-ruby/api_operations/page.rb (hunk 2 above)
Comment (gglin) [reply lib/clever-ruby/api_operations/page.rb]: ugh... should've read the documentation closer... I'll probably just do a new PR with that approach instead.

File: lib/clever-ruby/district_admin.rb (hunk 3 above)
Comment (gglin) [reply lib/clever-ruby/district_admin.rb:33]: I was wondering about this myself.  I think it makes more sense to trust whatever the API documentation / schema on their website says, rather than what's in the gem.  My guess is this is either an oversight w/ whoever wrote the gem initially, or their API changed from requiring name to requiring email.  I was thinking of submitting a PR to the official repo fixing that as well, but that's for another day

**Comment by gglin** [comment]
@ruddzw changes made to use `show_links=true` instead of standardizing the district_admin response manually. As far as the tests in the other PRs go, I think most of those were probably just b/c the VCR cassettes were outdated and needed updating?  Not exactly sure but I feel like updating `list_test` is probably enough beefing up.  We'll see I guess...

File: lib/clever-ruby/util.rb [hunk 4]
```diff
@@ -43,7 +43,7 @@ def self.convert_to_clever_object(resp)
           uri = resp[:links].select { |l| l[:rel] == 'self' }[0][:uri]
         end
 
-        klass_name = %r{/v1.1/([a-z_]+)/\S+$}.match(uri)[1]
+        klass_name = %r{v1.1/([a-z_]+)/\S+$}.match(uri)[1]
```
Comment (ruddzw) [inline lib/clever-ruby/util.rb:46]: Just curious why this change was needed

**Review by ruddzw — APPROVED** [review]
👍

File: lib/clever-ruby/util.rb (hunk 4 above)
Comment (gglin) [reply lib/clever-ruby/util.rb:46]: See full commit message in https://github.com/gglin/clever-ruby/pull/1/commits/c9475f1c45c754487ea17a6f23b68f624ced6ede.  We don't need this anymore since we're using `show_links`, but I figured this regex being less strict is still not a bad thing
